### PR TITLE
[#3163] Refactor(spark-connector): Refactor Spark-connector IT to support more iceberg catalog backends

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?
Refactor Spark-connector IT.

### Why are the changes needed?
to support more iceberg catalog backends, such as testing hive, jdbc, rest catalog backends.

Fix: https://github.com/datastrato/gravitino/issues/3163

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existing ITs.

### CODE REVIEW

**Comment by caican00** [comment]
Hi @FANNG1 could you help review this PR? Thank you.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/SparkEnvIT.java [hunk 1]
```diff
@@ -32,18 +30,20 @@ public abstract class SparkEnvIT extends SparkUtilIT {
...
 
-  protected FileSystem hdfs;
   private final String metalakeName = "test";
-
   private SparkSession sparkSession;
-  private String hiveMetastoreUri = "thrift://127.0.0.1:9083";
   private String gravitinoUri = "http://127.0.0.1:8090";
-  private String warehouse;
+
+  protected String hiveMetastoreUri = "thrift://127.0.0.1:9083";
```
Comment (FANNG1) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/SparkEnvIT.java:37]: could you place `protected` variable  before `private` ?

**Comment by FANNG1** [comment]
LGTM, except one comment

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/SparkEnvIT.java (hunk 1 above)
Comment (caican00) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/SparkEnvIT.java:37]: i would like to use `hiveMetastoreUri` in `SparkIcebergCatalogHiveBackendIT`.

Comment (caican00) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/SparkEnvIT.java:33]: done

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**